analysis/quantifier-comparisons/pre-processing-benchmarking-metrics.Rmd

```diff
@@ -0,0 +1,612 @@
+  )
+}
+```
+
+```{r}
+alevin_sces <- alevin_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+alevin_fry_sces <- alevin_fry_sces %>%
+  purrr::map(addPerCellQC_mito)
+
+alevin_fry_sketch_sces <- alevin_fry_sketch_sces %>%
+  purrr::map(addPerCellQC_mito)
+
+kallisto_sces <- kallisto_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+cellranger_sces <- cellranger_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+```
+
+
+```{r}
+## merge all QC output into one data frame to work with for plotting comparisons
+celldata_to_df <- function(sce){
+  # extract the column (cell) summary data from a SCE, 
+  # convert to data frame and move cell id to a column
+  as.data.frame(colData(sce)) %>%
+  tibble::rownames_to_column(var = "cell_id") %>%
+    # add a column to get # of cells detected in that sample
+    mutate(cells_detected = ncol(sce),
+           # add genes per cell for each sample
+           genes_detected = nrow(sce))
+}
+
+alevin_cell_qc <- alevin_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+alevin_fry_cell_qc <- alevin_fry_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+alevin_fry_sketch_cell_qc <- alevin_fry_sketch_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+kallisto_cell_qc <- kallisto_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+cellranger_cell_qc <- cellranger_sces %>%
+  purrr::map_df(celldata_to_df, .id = "quant_id")
+
+# combine all the data frames into one
+cell_qc <- dplyr::bind_rows(
+  alevin = alevin_cell_qc,
+  alevin_fry = alevin_fry_cell_qc,
+  alevin_fry_sketch = alevin_fry_sketch_cell_qc,
+  kallisto = kallisto_cell_qc,
+  cellranger = cellranger_cell_qc,
+  .id = "tool"
+) %>%
+  dplyr::left_join(quant_info,
+                   by = c("tool" = "tool", 
+                          "quant_id" = "quant_dir"))
+```
+
+
+
+## Comparison of QC Metrics across all tools
+
+### Per Cell QC
+
+Let's first start by taking a look at some general information across all of our tools and how consistent each of these metrics is for all four of the samples. Here are the metrics we want to look at: 
+
+  * Total number of cells detected
+  * Mitochondrial content/ cell
+  * Number of UMIs/cell
+  * Number of Genes/cell 
+
+
+```{r}
+## make new data frame with mean of cells detected and standard dev
+cells_detected_stats <- cell_qc %>%
+  group_by(sample, tool) %>%
+  summarise(cells_detected = mean(cells_detected)) %>%
+  group_by(tool) %>%
+  summarise(mean = mean(cells_detected),
+            standard_error = sd(cells_detected)/sqrt(length(cells_detected)))
+
+## first look at cells detected
+ggplot(cells_detected_stats, aes(x = tool,y = mean)) +
+  geom_bar(stat = "identity", fill = "darkblue") +
+  geom_errorbar(aes(x = tool, ymin=mean-standard_error, 
+                    ymax = mean + standard_error), 
+                width = 0.4, colour = "black", alpha = 0.9, size = 1.3)+
+  theme_classic() +
+  ylab("Total Cells Detected") + 
+  xlab("")
+```
```

Comment:
Since we don't have that many data points here (and even if we do, if I'm being really honest), let's not use bar plots here. I'd rather see all of the data with a jitter plot (or my favorite, [a sina plot](https://ggforce.data-imaginist.com/reference/geom_sina.html)) perhaps with an overlaid boxplot. For these exploratory plots, I am generally in favor of letting ggplot do the work, rather than trying to calculate summaries myself.

A second thought is that we can combine this with the next plot (or get the next plot's info out of this one) by using color for the points.
Something like this, perhaps (not tested).


```suggestion
cells_detected_stats <- cell_qc %>%
  group_by(sample, tool) %>%
  summarise(cells_detected = mean(cells_detected))

## first look at cells detected
ggplot(cells_detected_stats, aes(x = tool, y = cells_detected)) +
  geom_jitter(mapping = aes(color = sample)) +
  geom_boxplot(alpha = 0.6)
  theme_classic() +
  ylab("Total Cells Detected") + 
  xlab("")

```